src/schema-generator/conditional-render.ts

```diff
@@ -22,28 +25,69 @@ export const parseConditionalRenders = (
 	formValues: TypeOfShape<Assign<ObjectShape, ObjectShape>>,
 	yupContext: Yup.TestContext<AnyObject>
 ) => {
-	if (isEmpty(formValues)) return true;
+	const fieldsWithShownRule: TFieldsWithShownRule = {};
 	yupContext.schema.withMutation((schema: Yup.ObjectSchema<Assign<ObjectShape, ObjectShape>>) => {
 		let yupSchema: ObjectShape = yupContext.schema.clone().describe().meta.schema;
 		Object.values(sections).forEach((section) => {
-			yupSchema = parseChildrenConditionalRenders(section.children, yupSchema, formValues);
+			yupSchema = parseChildrenConditionalRenders(section.children, yupSchema, formValues, fieldsWithShownRule);
 		});
+		yupSchema = parseShownRule(sections, yupSchema, fieldsWithShownRule);
 		schema.shape(yupSchema);
 	});
 	return true;
 };
 
+/**
+ * Check if fields that depend on the shown rule can be rendered
+ * @param sections JSON representation of the fields
+ * @param yupSchema Generated Yup schema object
+ * @param fieldsWithShownRule Fields with shown rule and the corresponding dependencies
+ * @returns Yup schema containing only fields that pass shown rule
+ */
+const parseShownRule = (
+	sections: TSectionsSchema,
+	yupSchema: ObjectShape,
+	fieldsWithShownRule: TFieldsWithShownRule
+) => {
+	const parsedYupSchema = { ...yupSchema };
+
+	const order = getFieldSortOrder(sections);
+
+	Object.entries(fieldsWithShownRule)
+		.sort(([fieldIdA], [fieldIdB]) => order[fieldIdA] - order[fieldIdB])
+		.forEach(([fieldId, meta]) => {
+			const notShown = meta.sourceIds.every((rules) => {
+				if (rules.length) {
+					return rules.some((id) => !parsedYupSchema[id] || parsedYupSchema[id].describe().meta?.hidden);
+				}
+				return false;
+			});
+			if (notShown) {
+				const hiddenFieldIdList = [fieldId, ...meta.childIds];
+				hiddenFieldIdList.forEach((hiddenFieldId) => {
+					parsedYupSchema[hiddenFieldId] = Yup.mixed()
```

Comment:
this is triggering the `unhandled schema type for mixed` warning from yup-helper, the `mixed` type has not been handled because there was no use case for it. but now that we need to have it, let's add its handling